### ISSUE TITLE
[VPNKit] panic: tap2vif: read: read /dev/net/tun: not pollable

### ISSUE THREAD

**Issue by AkihiroSuda** [rootless-containers/rootlesskit#130]
```
panic: tap2vif: read: read /dev/net/tun: not pollable

goroutine 19 [running]:
github.com/rootless-containers/rootlesskit/pkg/network/vpnkit.tap2vif(0xc0001700a0, 0x7eff64137198, 0xc00000c200)
        /tmp/tmp.wZbjFCTJTg/src/github.com/rootless-containers/rootlesskit/pkg/network/vpnkit/vpnkit.go:195 +0x16f
created by github.com/rootless-containers/rootlesskit/pkg/network/vpnkit.startVPNKitRoutines
        /tmp/tmp.wZbjFCTJTg/src/github.com/rootless-containers/rootlesskit/pkg/network/vpnkit/vpnkit.go:185 +0x516
```

RootlessKit v0.9.2, `dockerd-rootless.sh` on moby/moby@37defbfd9b968f38e8e15dfa5f06d9f878bd65ba .
Happened during early startup phase.

**Comment by iHiD** [comment]
I also just hit the same issue. @AkihiroSuda Did you find a solution?

**Comment by AkihiroSuda** [comment]
This seems a solution, but haven't tested

https://github.com/golang/go/issues/38618#issuecomment-619387175

> ... using unix.Open instead of os.OpenFile and setting the fd to NonBlocking seems to solve the problem, ...

**Comment by AkihiroSuda** [comment]
A workaround is to use slirp4netns
https://github.com/rootless-containers/slirp4netns

Rootless Docker automatically switches from VPNKit to slirp4netns if slirp4netns (v0.4.0 or later) is installed.

**Comment by iHiD** [comment]
OK, great. Thanks. I'll try `slirp4netns` later.

**Comment by AkihiroSuda** [comment]
Should be fixed in RootlessKit v0.14.3
